### PR TITLE
add enable-shape-inference option for compilation.

### PR DESCRIPTION
**Pull request by xiaojiqing**
add enable-shape-inference option for compilation.

### CODE REVIEW

File: cc/compile_and_test_inner.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ function print_compile_options() {
     echo "enable_protocol_mpc_securenn: ${rtt_enable_protocol_mpc_securenn}"
     echo "   enable_protocol_mpc_helix: ${rtt_enable_protocol_mpc_helix}"
     echo "               enable_128bit: ${rtt_enable_128bit}"
+    echo "      enable_shape_inference: ${rtt_enable_shape_inference}"
```
Comment (vingstar) [inline cc/compile_and_test_inner.sh:30]: how about making it more adaptive and intelligent? Such that if the current python environment is '3.7', this option is default enabled.

File: cc/tf/rttops/math_ops.cc [hunk 2]
```diff
@@ -208,53 +215,64 @@ REGISTER_OP("RttAddN")
...
   .Output("res: string")
   .Attr("transpose_a: bool = false")
   .Attr("transpose_b: bool = false");
-  // .SetShapeFn(::tensorflow::shape_inference::MatMulShape);
 
 REGISTER_OP("RttSquare").Input("x: string").Output("res: string");
 
-
 REGISTER_OP("RttReduceMean")
+#if ROSETTA_ENABLES_SHAPE_INFERENCE
```
Comment (vingstar) [inline cc/tf/rttops/math_ops.cc:230]: Please move this into the end of this MACRO.

**Comment by ujnss** [comment]
@xiaojiqing missing a file, `rtt_completion`.

**Review by vingstar — CHANGES_REQUESTED** [review]
please refine it as instructed, for example the `if` clause should be in the end of REGISTER_OP.

File: cc/tf/rttops/math_ops.cc (hunk 2 above)
Comment (xiaojiqing) [reply cc/tf/rttops/math_ops.cc:230]: REGISTER_OP("RttReduceMean")
  .Input("input: string")
  .Input("reduction_indices: Tidx")
  .Output("output: string")
  .Attr("keep_dims: bool = false")
  .Attr("Tidx: {int32, int64} = DT_INT32")
#if ROSETTA_ENABLES_SHAPE_INFERENCE
  .SetShapeFn(::tensorflow::shape_inference::ReductionShape)
#endif
;
change it in this way?

File: cc/compile_and_test_inner.sh (hunk 1 above)
Comment (xiaojiqing) [reply cc/compile_and_test_inner.sh:30]: just add some shell code to check the environment and set the value according to the checking result.

File: cc/tf/rttops/math_ops.cc (hunk 2 above)
Comment (vingstar) [reply cc/tf/rttops/math_ops.cc:230]: yes

**Comment by vingstar** [comment]
LGTM, thank you!

**Pull request merged**